crates/yas_periphery/src/tests/test_yas_nft_position_manager.cairo

```diff
@@ -0,0 +1,243 @@
+mod YASNFTPositionManagerTests {
+    use starknet::{ClassHash, ContractAddress, SyscallResultTrait};
+    use starknet::syscalls::deploy_syscall;
+    use starknet::testing::{set_contract_address};
+    use integer::BoundedInt;
+
+    use yas_core::contracts::yas_erc20::{
+        ERC20, ERC20::ERC20Impl, IERC20Dispatcher, IERC20DispatcherTrait
+    };
+    use yas_core::contracts::yas_factory::{
+        YASFactory, IYASFactory, IYASFactoryDispatcher, IYASFactoryDispatcherTrait
+    };
+    use yas_core::libraries::tick_math::{TickMath::MIN_TICK, TickMath::MAX_TICK};
+    use yas_core::numbers::fixed_point::implementations::impl_64x96::{
+        FP64x96Impl, FixedType, FixedTrait
+    };
+    use yas_core::numbers::signed_integer::{
+        i32::i32, i32::i32_div_no_round, integer_trait::IntegerTrait
+    };
+    use yas_core::tests::utils::constants::FactoryConstants::{
+        POOL_CLASS_HASH, FeeAmount, fee_amount, tick_spacing, OWNER
+    };
+    use yas_core::tests::utils::constants::PoolConstants::{
+        TOKEN_A, TOKEN_B, POOL_ADDRESS, WALLET, encode_price_sqrt_1_1
+    };
+
+    use yas_periphery::yas_nft_position_manager::{
+        YASNFTPositionManager, IYASNFTPositionManager, IYASNFTPositionManagerDispatcher,
+        IYASNFTPositionManagerDispatcherTrait, MintParams
+    };
+
+    fn setup() -> (IYASNFTPositionManagerDispatcher, IERC20Dispatcher, IERC20Dispatcher) {
+        let yas_factory = deploy_factory(OWNER(), POOL_CLASS_HASH()); // 0x1
+        let nft_position_manager = deploy_nft_position_manager(yas_factory.contract_address); // 0x2
+
+        // Deploy ERC20 tokens with factory address
+        let token_0 = deploy_erc20('YAS0', '$YAS0', 4000000000000000000, OWNER()); // 0x3
+        let token_1 = deploy_erc20('YAS1', '$YAS1', 4000000000000000000, OWNER()); // 0x4
+
+        set_contract_address(OWNER());
+        token_0.transfer(WALLET(), 4000000000000000000);
+        token_1.transfer(WALLET(), 4000000000000000000);
```

Comment:
If the tests don't require a specific token limit, we could simply set it to `BoundedInt::max()` rather than the fixed value of 4x10¹⁸ . This approach was effective in resolving other issues in the past. Do you think this could apply here as well?

Reply:
✔️